src/uwtools/api/rocoto.py

```diff
@@ -1,18 +1,34 @@
+from typing import Union
+
+from uwtools.config.formats.yaml import YAMLConfig
 from uwtools.rocoto import realize_rocoto_xml as _realize
 from uwtools.rocoto import validate_rocoto_xml_file as _validate
 from uwtools.types import OptionalPath
 
 
-def realize(input_file: OptionalPath = None, output_file: OptionalPath = None) -> bool:
+def realize(config: Union[YAMLConfig, OptionalPath], output_file: OptionalPath = None) -> bool:
```

Comment:
Updated signature to be more flexible for API users (the CLI did not need this flexibility).